cpp/log/etcd_consistent_store-inl.h

```diff
@@ -314,6 +386,73 @@ std::string EtcdConsistentStore<Logged>::GetFullPath(
   return root_ + key;
 }
 
+
+// static
+template <class Logged>
+template <class T>
+Update<T> EtcdConsistentStore<Logged>::TypedUpdateFromWatchUpdate(
+    const EtcdClient::WatchUpdate& update) {
+  const std::string raw_value(util::FromBase64(update.node_.value_.c_str()));
+  T thing;
+  CHECK(thing.ParseFromString(raw_value)) << raw_value;
+  EntryHandle<T> handle(thing);
+  if (update.exists_) {
+    handle.SetHandle(update.node_.modified_index_);
+  }
+  return Update<T>(handle, update.exists_);
+}
+
+
+template <class Logged>
+void EtcdConsistentStore<Logged>::MaybeUpdateLocalServingSTH(
+    const EntryHandle<ct::SignedTreeHead>& handle) {
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (!serving_sth_ ||
+      serving_sth_->Entry().timestamp() < handle.Entry().timestamp()) {
+    VLOG(1) << "Updating serving_sth_ to: " << handle.Entry().DebugString();
+    serving_sth_.reset(new EntryHandle<ct::SignedTreeHead>(handle));
+  } else {
+    VLOG(1) << "Not updating serving_sth: "
+            << serving_sth_->Entry().timestamp()
+            << ">=" << handle.Entry().timestamp();
+  }
+}
+
+
+template <class Logged>
+void EtcdConsistentStore<Logged>::OnEtcdServingSTHUpdated(
+    Notification* notify,
+    const std::vector<EtcdClient::WatchUpdate>& updates) {
+  CHECK_LE(updates.size(), 1);
+
+  if (!updates.empty()) {
+    const Update<ct::SignedTreeHead> update(
+        TypedUpdateFromWatchUpdate<ct::SignedTreeHead>(updates[0]));
+    MaybeUpdateLocalServingSTH(update.handle_);
```

Comment:
Yeah, tbh the updates in `SetServingSTH` were really only being useful in testing, and as you correctly picked up on I really only wanted to update with etcd's view of the world.
Fixed.